### ISSUE TITLE
Unify love-android Dependencies.

### ISSUE THREAD

**Issue by MikuAuahDark** [love2d/megasource#3]
I think it's a good idea to unify the dependencies of love-android into Megasource. However there are some things that needs some attention.

- [x] This proposal must be approved first, at least by slime.
- [ ] libtheora needs to be updated to 1.2.0-alpha1. The CMake script can be adopted from the Android.mk
- [x] LuaJIT needs to be updated to at least LuaJIT/LuaJIT@e9af1abec542e6f9851ff2368e7f196b6382a44c for compatibility with [HWASan](https://source.android.com/devices/tech/debug/hwasan).
- [x] ~~OpenAL-soft needs to be updated to 1.21.1 and Oboe 1.4.3 needs to be placed beside OpenAL-soft folder.~~ Not necessary anymore.
- [x] There are some CMake modifications needed both to Megasource and LOVE.

If this proposal is accepted, I'd love to see it in LOVE 12.0. As a side note, Android.mk is no longer needed once the deps are unified, which means LOVE Android will use CMake as its build system.

**Comment by slime73** [comment]
Sounds reasonable to me, as long as it won't make it significantly harder to support changes to Android's build setup in future Android versions.

**Comment by MikuAuahDark** [comment]
Only Theora remains, but considering Theora development stopped, I don't think it's worth the effort to update to 1.2.0-alpha1.
